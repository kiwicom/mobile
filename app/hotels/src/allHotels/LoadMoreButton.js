// @flow

import * as React from 'react';
import { View } from 'react-native';
import {
  IconLoading,
  LinkButton,
  StyleSheet,
} from '@kiwicom/react-native-app-shared';
import Translation from '@kiwicom/react-native-app-translations';

type Props = {|
  isLoading: boolean,
  onPress: () => void,
|};

const styleSheet = StyleSheet.create({
  wrapper: {
    height: 50,
    justifyContent: 'center',
  },
});

/**
 * This button is used as a "Load More" button at the end of the list of all hotels.
 */
export default function LoadMoreButton({ isLoading, onPress }: Props) {
  return (
    <View style={styleSheet.wrapper}>
      {isLoading ? (
        <IconLoading />
      ) : (
        <LinkButton
          title={<Translation id="hotels_search.load_more" />}
          onPress={onPress}
        />
      )}
    </View>
  );
}
