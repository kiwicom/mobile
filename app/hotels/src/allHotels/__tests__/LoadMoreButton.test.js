// @flow

import * as React from 'react';
import { PlaygroundRenderer } from '@kiwicom/mobile-playground';

import LoadMoreButton from '../LoadMoreButton';

const VoidAction = () => {};

it('renders the buttons in the playground', () => {
  PlaygroundRenderer.render(<LoadMoreButton isLoading={true} onPress={VoidAction} />);
  PlaygroundRenderer.render(<LoadMoreButton isLoading={false} onPress={VoidAction} />);
});
