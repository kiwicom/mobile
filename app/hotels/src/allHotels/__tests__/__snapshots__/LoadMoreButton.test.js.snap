// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the buttons in the playground 1`] = `
<View
  style={
    Object {
      "height": 50,
      "justifyContent": "center",
    }
  }
>
  <IconLoading
    size="small"
  />
</View>
`;

exports[`renders the buttons in the playground 2`] = `
<View
  style={
    Object {
      "height": 50,
      "justifyContent": "center",
    }
  }
>
  <LinkButton
    onPress={[Function]}
    title={
      <Translation
        id="hotels_search.load_more"
      />
    }
  />
</View>
`;
